buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -184,6 +185,33 @@ class PerformanceTestPlugin : Plugin<Project> {
             // Never up-to-date since it reads data from the database.
             outputs.upToDateWhen { false }
         }
+
+        createPerformanceDefinitionJsonTask("writePerformanceScenarioDefinitions", performanceSourceSet, true)
+        createPerformanceDefinitionJsonTask("verifyPerformanceScenarioDefinitions", performanceSourceSet, false)
+    }
+
+    /**
+     * Scan performance test scenarios and:
+     * 1. Write to the scenario JSON file if write=true;
+     * 2. Only verify the scenario JSON file is correct if write=false.
+     */
+    private
+    fun Project.createPerformanceDefinitionJsonTask(name: String, performanceSourceSet: SourceSet, write: Boolean) {
+        val outputJson = project.rootProject.file(".teamcity/performance-tests-ci.json")
+        tasks.register<Test>(name) {
```

Comment:
Given that we won't run the write task on CI, I think it is fine to make this task non-cacheable with all the extra hoops we would need to jump. Let's keep the up-to-date checks, though.